src/TagsInput/TagsInput.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+
+import * as React from 'react';
+import { View, Platform } from 'react-native';
+import type { ViewLayoutEvent } from 'react-native/Libraries/Components/View/ViewPropTypes';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+
+import DeleteButton from './components/DeleteButton';
+import TagsContainer from './components/TagsContainer';
+import InputField from './components/InputField';
+import { StyleSheet } from '../PlatformStyleSheet';
+import { Text } from '../Text';
+
+import { MIN_WIDTH } from './constants';
+
+type Props = {|
+  +label: string,
+  +onChangeText: (value: string) => void,
+  +placeholder: string,
+  +selected?: string[],
+  +disabled?: boolean,
+  +fontSize: number,
+  +onClearPress?: () => void,
+  +value?: string,
+|};
+
+type State = {
+  value: string,
+  containerWidth: ?number,
+  inputWidth: ?number,
+};
+
+class TagInput extends React.Component<Props, State> {
+  inputRef: any; // @TODO find a way how to type this properly, the issue with mixing react and react-native
+  static defaultProps = {
+    fontSize: 16,
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      value: props.value ?? '',
+      containerWidth: null,
+      inputWidth: null,
+    };
+  }
+
+  componentDidUpdate = (prevProps: Props) => {
+    const { value } = this.props;
+    if (value !== prevProps.value) {
+      this.setState({ value });
+    }
+  };
+
+  setContainerWidth = (event: ViewLayoutEvent) => {
+    this.setState({ containerWidth: event.nativeEvent.layout.width });
+  };
+
+  setInputWidth = (event: ViewLayoutEvent) => {
+    this.setState({ inputWidth: event.nativeEvent.layout.width });
+  };
+
+  setRef = (input: React.Element<any>) => {
+    this.inputRef = input;
+  };
+
+  getWidth = (width: ?number) => {
+    const { containerWidth } = this.state;
+    if (containerWidth && width) {
+      return containerWidth - width;
+    }
+    return undefined;
```

Comment:
I've always been told that it was better to return `null` than `undefined`, as `null` meant value could not be set instead of being undefined.

Reply:
OK, just there is another side of the coin, https://youtu.be/rhV6hlL_wMc?t=1326 😄 
But I will as you suggest.

Reply:
Thanks for the reference, I think he's got some very valid points. If we feel strongly about this, then we should enforce it.